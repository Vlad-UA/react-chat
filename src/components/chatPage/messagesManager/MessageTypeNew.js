import React from 'react';
import classNames from 'classnames';
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Input from 'material-ui/Input';
import Button from 'material-ui/Button';
import PropTypes from 'prop-types';

const styles = theme => ({
  root: {
    margin: 20,
  },
  input: {
    margin: theme.spacing.unit,
    width: '94%',
    marginTop: 12,
    left: 20,
  },
  joinButton: {
    marginLeft: 15,
    width: '95%',
  },
});

class MessageTypeNew extends React.Component {
  state = {
    value: '',
  };

  handleKeyPress = (event) => {
    const { value } = this.state;

    if (event.key === 'Enter' && value) {
      this.props.sendMessage(value);
      this.setState({ value: '' });
    }
  };

  handleValueChange = (event) => {
    this.setState({
      value: event.target.value,
    });
  };

  render() {
    const {
      classes, classAdditional, showJoinButton, onJoinButtonClick, disabled,
    } = this.props;

    return (
      <Paper className={classNames(classAdditional, classes.root)} elevation={4}>
        {showJoinButton ? (
          <Button
            disabled={disabled}
            fullWidth
            variant="raised"
            color="primary"
            onClick={onJoinButtonClick}
            className={classes.joinButton}
          >
            Join
          </Button>
        ) : (
          <Input
            disabled={disabled}
            placeholder="Type your message..."
            className={classes.input}
            onKeyPress={this.handleKeyPress}
            onChange={this.handleValueChange}
            value={this.state.value}
          />
        )}
      </Paper>
    );
  }
}

MessageTypeNew.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  classAdditional: PropTypes.string,
  sendMessage: PropTypes.func.isRequired,
  onJoinButtonClick: PropTypes.func.isRequired,
  showJoinButton: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
};
MessageTypeNew.defaultProps = {
  classAdditional: '',
};

export default withStyles(styles)(MessageTypeNew);
