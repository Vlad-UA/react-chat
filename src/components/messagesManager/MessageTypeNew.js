import React from 'react';

import {withStyles} from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Input from 'material-ui/Input';

const styles = theme => ({
  input: {
    margin: theme.spacing.unit,
    width: '100%',
    marginTop: 12
  },
});

function MessageTypeNew(props) {
  const {classes, classAdditional} = props;
  return (
    <Paper className={classAdditional} elevation={4}>
      <Input
        placeholder="Type your message..."
        className={classes.input}
      />
    </Paper>
  );
}

export default withStyles(styles)(MessageTypeNew);
