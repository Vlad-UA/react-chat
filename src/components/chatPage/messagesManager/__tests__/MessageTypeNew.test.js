/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageTypeNew from '../MessageTypeNew';

const mockProps = {
  sendMessage: jest.fn(),
  onJoinButtonClick: jest.fn(),
  showJoinButton: true,
  disabled: false,
};

describe('<MessageTypeNew />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageTypeNew {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot renders correctly - Join to chat button (enabled)', () => {
    const tree = renderer
      .create(<MessageTypeNew {...mockProps} showJoinButton disabled={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - Join to chat button (disabled)', () => {
    const tree = renderer
      .create(<MessageTypeNew {...mockProps} showJoinButton disabled />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - Input message field (enabled)', () => {
    const tree = renderer
      .create(<MessageTypeNew {...mockProps} showJoinButton={false} disabled={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - Input message field (disabled)', () => {
    const tree = renderer
      .create(<MessageTypeNew {...mockProps} showJoinButton={false} disabled />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
