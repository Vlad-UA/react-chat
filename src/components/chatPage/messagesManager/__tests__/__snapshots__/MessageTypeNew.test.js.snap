// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageTypeNew /> snapshot renders correctly - Input message field (disabled) 1`] = `
<div
  className="MuiPaper-root-59 MuiPaper-elevation4-65 MuiPaper-rounded-60 MessageTypeNew-root-56"
>
  <div
    className="MuiInput-root-111 MuiInput-disabled-114 MuiInput-underline-115 MessageTypeNew-input-57"
    onKeyPress={[Function]}
  >
    <input
      aria-invalid={undefined}
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-119 MuiInput-disabled-114"
      defaultValue={undefined}
      disabled={true}
      id={undefined}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Type your message..."
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<MessageTypeNew /> snapshot renders correctly - Input message field (enabled) 1`] = `
<div
  className="MuiPaper-root-59 MuiPaper-elevation4-65 MuiPaper-rounded-60 MessageTypeNew-root-56"
>
  <div
    className="MuiInput-root-111 MuiInput-underline-115 MessageTypeNew-input-57"
    onKeyPress={[Function]}
  >
    <input
      aria-invalid={undefined}
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-119"
      defaultValue={undefined}
      disabled={false}
      id={undefined}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Type your message..."
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<MessageTypeNew /> snapshot renders correctly - Join to chat button (disabled) 1`] = `
<div
  className="MuiPaper-root-59 MuiPaper-elevation4-65 MuiPaper-rounded-60 MessageTypeNew-root-56"
>
  <button
    className="MuiButtonBase-root-101 MuiButtonBase-disabled-102 MuiButton-root-86 MuiButton-raised-91 MuiButton-raisedPrimary-92 MuiButton-disabled-95 MuiButton-fullWidth-100 MessageTypeNew-joinButton-58"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-87"
    >
      Join
    </span>
  </button>
</div>
`;

exports[`<MessageTypeNew /> snapshot renders correctly - Join to chat button (enabled) 1`] = `
<div
  className="MuiPaper-root-59 MuiPaper-elevation4-65 MuiPaper-rounded-60 MessageTypeNew-root-56"
>
  <button
    className="MuiButtonBase-root-101 MuiButton-root-86 MuiButton-raised-91 MuiButton-raisedPrimary-92 MuiButton-fullWidth-100 MessageTypeNew-joinButton-58"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-87"
    >
      Join
    </span>
    <span
      className="MuiTouchRipple-root-104"
    />
  </button>
</div>
`;
